src/utils.js

```diff
@@ -22,6 +24,42 @@ const isInvalidValidPassword = password => {
   }
 };
 
+/**
+ * @desc get hash from dApp
+ * @param {object} content
+ * @return {string} hash string
+ */
+const hashDappContent = content => {
+  let buff = Buffer.allocUnsafe(0);
+
+  const { name = {} } = content;
+  // 1.append name into buffer
+  //sort keys in increasing order
+  const nameLanguagesKey = Object.keys(name).sort();
+  for (let index = 0; index < nameLanguagesKey.length; index++) {
+    //append language code
+    const key = nameLanguagesKey[index];
+    buff = Buffer.concat([buff, Buffer.from(key, 'utf8')]);
```

Comment:
I think the Buffer has a `write()` method which would be more clear than re assigning it all the time. It would be nice if you can use write. If it doesn't work it's ok - more a nice to have

Reply:
The buffer in nodejs is fixed size, so write() only work to override the content of buffer, you need concat and reassign the buffer to new buffer with bigger size

Reply:
ok got it